api.py

```diff
@@ -0,0 +1,116 @@
+import os
+import threading
+import enum
+from time import sleep
+from subprocess import Popen, DEVNULL
+
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+
+from pytube import YouTube
+from pytube.exceptions import AgeRestrictedError, VideoUnavailable, RegexMatchError
+
+
+class State(enum.Enum):
+    INTERLUDE = "interlude"
+    PLAYING = "playing"
+
+app = FastAPI()
+interlude_command = ['ffmpeg', '-re', '-stream_loop', '-1', '-i', "./interlude.mp4", '-vf', 
+                     'scale=426:240', '-c:v', 'libx264', 
+                     '-preset', 'veryfast', '-tune', 'zerolatency', 
+                     '-c:a', 'aac', '-ar', '44100', '-f', 'flv', 
+                     'rtmp://localhost:1935/live/mystream']
+process_dict = {}
+process_dict[State.INTERLUDE] = Popen(interlude_command, stdout=DEVNULL, stdin=DEVNULL, stderr=DEVNULL)
+current_video_data = {}
+
+# Ensure video folder exists
+if not os.path.exists("./videos"):
+   os.makedirs("./videos")
+
+# Enable CORS
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+def download_and_play(url:str):
+    # Download video
+    video = YouTube(url)
+    video = video.streams.get_by_resolution("360p")
```

Comment:
lets adjust this to ensure we get a 360p video combined with sound:
```suggestion
    video = video.streams.filter(
                    resolution="360p",
                    progressive=True,
                  )
                  .order_by("resolution")
                  .desc()
                  .first()
```

see this stackoverflow post on why `progressive=True` is cool https://stackoverflow.com/a/63223757